src/mockServiceWorker.js

```diff
@@ -74,11 +74,22 @@ self.addEventListener('message', async function (event) {
   }
 })
 
-self.addEventListener('fetch', async function (event) {
+self.addEventListener('fetch', function (event) {
   const { clientId, request } = event
   const requestClone = request.clone()
   const getOriginalResponse = () => fetch(requestClone)
 
+  // Bypass navigation requests.
+  if (request.mode === 'navigate') {
+    return
+  }
+
+  // Bypass mocking if the current client isn't present in the internal clients map
+  // (i.e. has the mocking disabled).
+  if (!clients[clientId]) {
+    return
```

Comment:
When the client hasn't yet sent the `MOCK_ACTIVATE` event bypass early as well. Requests made between the worker registration and worker activation are performed as-is. This includes static assets as well, they won't be marked as XHR in this time window.

Note that the time window is subjected to a race condition, as worker activation and registration are both asynchronous actions and may happen with a different time distance.